src/Service/ValidationErrorHandler.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace App\Service;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\Persistence\Proxy;
+use Symfony\Component\Config\Definition\Exception\InvalidTypeException;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+class ValidationErrorHandler
+{
+    public function __construct(
+        private EntityManagerInterface $entityManager,
+        private ValidatorInterface $validator,
+    ) {
+    }
+
+    public function getErrors($entity): array
+    {
+        if (!$this->isEntity($entity)) {
+            throw new InvalidTypeException('Non-entity type was passed to validation service');
+        }
+
+        $errors = $this->validator->validate($entity);
+
+        $messages = [];
+        if (count($errors) > 0) {
```

Comment:
Not sure how useful this service is.
You are getting errors from the validator here, looping to generate an array. Then, you are looping through that array in the controller to add to the form